src/components/trip_view/cal/CalViewOptions.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react';
+import { Button, ButtonGroup } from 'react-bootstrap';
+import queryString from 'query-string';
+import PropTypes from 'prop-types';
+
+/**
+ * This component has all of the options for routing that allow the
+ * TripCal to show the right elements.
+ * It routes to addresses such as '/trips?id=29&view=month', which
+ * means trip id 29, with the month view open.
+ */
+export class CalViewOptions extends Component {
+  constructor(props) {
+    super(props);
+    this.getPath = this.getPath.bind(this);
+    this.getQuery = this.getQuery.bind(this);
+    this.routeTo = this.routeTo.bind(this);
+  }
+
+  /** Get the path of the current browser's URL */
+  getPath() {
+    return this.props.history.location.pathname;
```

Comment:
Ahhh yupp, here we go. Again, awesome.

Reply:
I'm assuming this is also related to the discussion on TripView's history object? 

If this is a significant issue, I'm in favour of integrating [this](https://www.npmjs.com/package/history) library into our application.

Reply:
I like the look of that library! I've put it into issue #159 (might not be done for this milestone, but it will for the next one).